src/metabase/lib/schema/expression/temporal.cljc

```diff
@@ -34,19 +41,35 @@
                               :get-day :get-day-of-week
                               :get-month :get-quarter :get-year}]
   (mbql-clause/define-tuple-mbql-clause temporal-extract-op :- :type/Integer
-  #_:datetime [:schema [:ref ::expression/temporal]]))
-
-(mbql-clause/define-tuple-mbql-clause :get-week :- :type/Integer
-  #_:datetime [:schema [:ref ::expression/temporal]]
-  ;; TODO should this be in the options map?
-  #_:mode [:maybe [:enum :iso :us :instance]])
-
-(mbql-clause/define-tuple-mbql-clause :convert-timezone
-  #_:datetime [:schema [:ref ::expression/temporal]]
-  ;; TODO could be better specified - perhaps with a build time macro to inline the timezones?
-  ;; NOT expressions?
-  #_:target [:string]
-  #_:source [:maybe [:string]])
+    #_:datetime [:schema [:ref ::expression/temporal]]))
+
+(mr/def ::get-week-mode
+  [:enum :iso :us :instance])
+
+(mbql-clause/define-catn-mbql-clause :get-week :- :type/Integer
+  [:datetime [:schema [:ref ::expression/temporal]]]
+  ;; TODO : the mode should probably go in the options map in modern MBQL rather than have it be a separate positional
+  ;; argument. But we can't refactor everything in one go, so that will have to be a future refactor.
+  [:mode     [:? [:schema [:ref ::get-week-mode]]]])
+
+(mr/def ::timezone-id
+  [:and
+
+   ::common/non-blank-string
+   (into [:enum
+          {:error/message "valid timezone ID"
+           :error/fn      (fn [{:keys [value]} _]
+                            (str "invalid timezone ID: " (pr-str value)))}]
```

Comment:
I added a custom `:error/fn` because otherwise it prints out all 600 valid options which is just too much.